### PR TITLE
feat(mocknet): improvements

### PR DESCRIPTION
**Pull request by wacban**
- added --yes to skip the user confirmation in reset
- added update-binaries command to have neard-runner re-download the neard binaries
- reworked update-config to work for arbitrary json configs

### CODE REVIEW

File: pytest/tests/mocknet/helpers/neard_runner.py [hunk 1]
```diff
@@ -172,15 +174,20 @@ def parse_binaries_config(self):
...
             binaries.append({
-                'url': b['url'],
-                'epoch_height': b['epoch_height'],
-                'system_path': self.home_path('binaries', f'neard{i}')
+                'url':
+                b['url'],
+                'epoch_height':
+                b['epoch_height'],
+                'system_path':
+                self.home_path('binaries', f'neard{i}')
```
Comment (wacban) [inline pytest/tests/mocknet/helpers/neard_runner.py:190]: Some auto formatting kicked in, do we have a recommended formatter?

File: pytest/tests/mocknet/mirror.py [hunk 2]
```diff
@@ -441,9 +451,19 @@ def start_traffic_cmd(args, traffic_generator, nodes):
...
+        help='''
+        A key value pair to set in the config. The key will be interpreted as a
+        json path to the config to be updated. The value will be parsed as json.   
+        e.g.
+        --set 'aaa.bbb.ccc=5'
+        --set 'aaa.bbb.ccc="5"'
+        --set 'aaa.bbb.ddd={"eee":6,"fff":"7"}' # no spaces!
+        ''',
+    )
+    update_config_parser.set_defaults(func=update_config_cmd)
```
Comment (VanBarbascu) [inline pytest/tests/mocknet/mirror.py:465]: Duplicated line.

File: pytest/tests/mocknet/helpers/neard_runner.py [hunk 3]
```diff
@@ -376,22 +390,24 @@ def do_network_init(self,
...
+
+            object = config
+            for key_item in key_item_list[:-1]:
+                if key_item not in object:
+                    object[key_item] = {}
+                object = object[key_item]
+
+            value = json.loads(value)
+
+            object[key_item_list[-1]] = value
```
Comment (VanBarbascu) [inline pytest/tests/mocknet/helpers/neard_runner.py:410]: Consider using [JOSNPath](https://www.digitalocean.com/community/tutorials/python-jsonpath-examples)

**Review by VanBarbascu — COMMENTED** [review]
I do not like the random style changes. We need a consistent style to avoid this clutter in the PR.

File: pytest/tests/mocknet/mirror.py [hunk 4]
```diff
@@ -419,6 +421,14 @@ def start_traffic_cmd(args, traffic_generator, nodes):
     )
 
 
+def neard_runner_update_binaries(node):
+    neard_runner_jsonrpc(node, 'update_binaries')
+
+
+def update_binaries_cmd(args, traffic_generator, nodes):
+    pmap(neard_runner_update_binaries, nodes)
```
Comment (marcelo-gonzalez) [inline pytest/tests/mocknet/mirror.py:429]: should it also redownload on the traffic generator? with: `pmap(neard_runner_update_binaries, nodes + [traffic_generator])`

File: pytest/tests/mocknet/helpers/neard_runner.py (hunk 1 above)
Comment (marcelo-gonzalez) [reply pytest/tests/mocknet/helpers/neard_runner.py:190]: not sure, but following what is done in `scripts/formatting`, I usually run: `python3 -m yapf -pir {filename}`

**Review by marcelo-gonzalez — APPROVED** [review]

File: pytest/tests/mocknet/helpers/neard_runner.py (hunk 3 above)
Comment (wacban) [reply pytest/tests/mocknet/helpers/neard_runner.py:405]: forgive me, I'm too lazy right now, and mocknet is going away anyway...

**Pull request merged**